cmd/elementGen/elements.go

```diff
@@ -0,0 +1,727 @@
+	}
+	sort.Strings(names)
+
+	var ele []Desc
+
+	for _, n := range names {
+		v := elements[n]
+		v.Name = n
+		v.JS = strings.ToLower(n)
+		if v.Override != "" {
+			v.Name = v.Override
+		}
+
+		v.DOM = fmt.Sprintf("HTML%sElement", v.Name)
+		if v.Basic {
+			v.DOM = "BasicHTMLElement"
+		}
+
+		for i := 0; i < len(v.Attributes); i++ {
+			a := &v.Attributes[i]
+			a.JS = strings.ToLower(a.Name)
+			if a.Type == "" {
+				a.Type = "string"
+			}
+		}
+
+		ele = append(ele, v)
+	}
+
+	executeTemplate("gen_elements_elementGen.go", primary, ele)
+	executeTemplate("gen_elements_elementGen_test.go", test, ele)
+}
+
+func executeTemplate(filename string, t *template.Template, ele []Desc) {
+	var b bytes.Buffer
+
+	if err := t.Execute(&b, ele); err != nil {
+		panic(err)
+	}
+
+	formatted, err := format.Source(b.Bytes())
+	if err != nil {
+		panic(err)
+	}
+
+	f, err := os.Create(filepath.Join(*outputDirectory, filename))
+	if err != nil {
+		panic(err)
+	}
+
+	if _, err := f.Write(formatted); err != nil {
+		panic(err)
+	}
+}
+
+const (
+	primaryTemplate = `
+	// Copyright (c) 2018 Paul Jolly <paul@myitcv.org.uk>, all rights reserved.
+	// Use of this document is governed by a license found in the LICENSE document.
+
+	// Code generated by elementGen. DO NOT EDIT.
+
+	package react
+
+	{{ range . }}
+	// {{ .Name }}Elem is the React element definition corresponding to the HTML <{{ .Name }}> element.
+	type {{ .Name }}Elem struct {
+		Element
+	}
+
+	// _{{ .Name }}Props defines the properties for the <{{ .Name }}> element.
+	type _{{ .Name }}Props struct {
+		*BasicHTMLElement	{{ if gt (len .Attributes) 0 }}
+
+		{{ range .Attributes }}{{ .Name }} {{ .Type }} ` + "`js:\"{{ .JS }}\"`" + `
+		{{ end }}
+		{{ end }}
+	}
+
+	// A creates a new instance of a <{{ .JS }}> element with the provided props and children.
+	func {{ .Name  }}(props *{{ .Name }}Props, children ...Element) *{{ .Name }}Elem {
+		rProps := &_{{ .Name }}Props{
+			BasicHTMLElement: newBasicHTMLElement(),
+		}
+
+		if props != nil {
+			props.assign(rProps)
+		}
+
+		return &{{ .Name }}Elem{
+			Element: createElement("{{ .JS }}", rProps, children...),
+		}
+	}
+	{{ end }}
+`
+	testTemplate = `
+	// +` + "build js" + `
+
+	// Code generated by elementGen. DO NOT EDIT.
+
+	package react_test
+
+	import (
+		"testing"
+
+		"honnef.co/go/js/dom"
+
+		"myitcv.io/react"
+		"myitcv.io/react/testutils"
+	)
+
+	{{ range . }}
+	func Test{{ .Name }}Elem(t *testing.T) {
+		class := "test"
+
+		x := testutils.Wrapper(react.{{ .Name  }}(&react.{{ .Name }}Props{ClassName: class}))
+		cont := testutils.RenderIntoDocument(x)
+
+		el := testutils.FindRenderedDOMComponentWithClass(cont, class)
+
+		if _, ok := el.(*dom.{{ .DOM }}); !ok {
+			t.Fatal("Failed to find <{{ .JS }}> element")
+		}
+	}
+	{{ end }}
+`
+)
+
+var (
+	// elements contains all of the Go wrappers to generate for the underlying HTML elements.
+	// Commented items have already been hand-written.
+	elements = map[string]Desc{
+		// "a"
```

Comment:
Is this comment a placeholder for replace the current `<a>` definition? Any reason we wouldn't do this now?